crates/bevy_audio/src/audio_output.rs

```diff
@@ -102,26 +103,30 @@ pub(crate) fn play_queued_audio_system<Source: Asset + Decodable>(
         (
             Entity,
             &Handle<Source>,
+            Option<&AudioTarget>,
             &PlaybackSettings,
             Option<&GlobalTransform>,
         ),
-        (Without<AudioSink>, Without<SpatialAudioSink>),
+        Or<((Without<AudioSink>, Without<SpatialAudioSink>), Changed<AudioTarget>)>,
     >,
     ear_positions: EarPositions,
     default_spatial_scale: Res<DefaultSpatialScale>,
     mut commands: Commands,
 ) where
     f32: rodio::cpal::FromSample<Source::DecoderItem>,
 {
-    let Some(stream_handle) = audio_output.stream_handle.as_ref() else {
+    let Some(global_stream_handle) = audio_output.stream_handle.as_ref() else {
         // audio output unavailable; cannot play sound
         return;
     };
 
-    for (entity, source_handle, settings, maybe_emitter_transform) in &query_nonplaying {
+    for (entity, source_handle, target, settings, maybe_emitter_transform) in &query_nonplaying {
         let Some(audio_source) = audio_sources.get(source_handle) else {
             continue;
         };
+
+        let stream_handle = target.as_ref().map(|target| &target.output_stream).unwrap_or(global_stream_handle);
```

Comment:
A comment above this line would be helpful: the intent isn't immediately clear without accessing to the PR description.